src/ParallelAlgorithms/Interpolation/Callbacks/ObserveSurfaceData.hpp

```diff
@@ -121,6 +238,28 @@ struct ObserveSurfaceData
         std::vector<ElementVolumeData>{{surface_name, tensor_components,
                                         extents_vector, bases_vector,
                                         quadratures_vector}});
+
+    std::vector<std::string> ylm_legend;
+    std::vector<double> ylm_data;
+    // The number of coefficients written will be (l_max + 1)^2 where l_max is
+    // the current value of l_max for this surface's strahlkorper. Because l_max
+    // remains constant, the number of coefficient columns written does, too. In
+    // the future when l_max is adaptive, instead of passing in the current
+    // l_max of the strahlkorper, we could pass in the maximum value that l_max
+    // could be to ensure that we (a) have enough columns to write all the
+    // coefficients regardless of the current value of l_max and (b) write a
+    // constant number of columns for each row of data regardless of the current
+    // l_max.
+    detail::fill_ylm_legend_and_data(make_not_null(&ylm_legend),
+                                     make_not_null(&ylm_data), strahlkorper,
+                                     time, strahlkorper.l_max());
+
+    const std::string ylm_subfile_name{std::string{"/"} + surface_name +
+                                       "_Ylm"};
+
+    Parallel::threaded_action<
+        observers::ThreadedActions::WriteReductionDataRow>(
+        proxy[0], ylm_subfile_name, ylm_legend, std::make_tuple(ylm_data));
```

Comment:
`std::move(ylm_legend)` and `std::move(ylm_data)` (not sure if that avoids a copy but can't hurt I think)